### PR TITLE
Add ad customizer example

### PR DESCRIPTION
**Pull request by Raibaz**

### CODE REVIEW

File: examples/AdvancedOperations/AddAdCustomizer.php [hunk 1]
```diff
@@ -0,0 +1,529 @@
...
+                PHP_EOL
+            );
+            $feedItemResourceNames []= $result->getResourceName();
+        }
+
+        return $feedItemResourceNames;
+    }
+
+    /**
+     * Helper function to create a FeedItemOperation.
```
Comment (fiboknacky) [inline examples/AdvancedOperations/AddAdCustomizer.php]: Nit: "Helper function" is redundant. Just focus on the actions in this method.

File: examples/AdvancedOperations/AddAdCustomizer.php [hunk 2]
```diff
@@ -0,0 +1,529 @@
...
+     *
+     * @param string name the value of the Name attribute.
+     * @param string price the value of the Price attribute.
+     * @param string date the value of the Date attribute.
+     * @param feedResourceName the resource name of the feed.
+     * @param feedAttributes the attributes to be set on the feed.
+     * @return a FeedItemOperation to create a feed item.
```
Comment (fiboknacky) [inline examples/AdvancedOperations/AddAdCustomizer.php]: Should have no "a". It'll break some IDEs.
After @return it should be only just a valid type.

Comment (fiboknacky) [reply examples/AdvancedOperations/AddAdCustomizer.php]: And PHPDoc is also wrong here.

File: examples/AdvancedOperations/AddAdCustomizer.php [hunk 3]
```diff
@@ -0,0 +1,529 @@
...
+     */
+    private function createFeedItems(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        string $adCustomizerFeedResourceName,
+        array $adCustomizerFeedAttributes
+    ) {
+        $feedItemOperations = [];
+        
+        $marsDate = new DateTime('first day of this month');
```
Comment (fiboknacky) [inline examples/AdvancedOperations/AddAdCustomizer.php]: $marsDate and $venusDate are used once.
Please do them inline.

File: examples/AdvancedOperations/AddAdCustomizer.php [hunk 4]
```diff
@@ -0,0 +1,529 @@
...
+        $feedItemOperations []= self::createFeedItemOperation(
+            "Mars",
+            "$1234.56",
+            date_format($marsDate, "Ymd His"),
+            $adCustomizerFeedResourceName,
+            $adCustomizerFeedAttributes
+        );
+
+        $venusDate = DateTime::createFromFormat('d', '15');
```
Comment (fiboknacky) [inline examples/AdvancedOperations/AddAdCustomizer.php]: And add a comment to explain what is 'd' and '15' here.

File: examples/AdvancedOperations/AddAdCustomizer.php [hunk 5]
```diff
@@ -0,0 +1,529 @@
...
+    private function createFeedItems(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        string $adCustomizerFeedResourceName,
+        array $adCustomizerFeedAttributes
+    ) {
+        $feedItemOperations = [];
+        
+        $marsDate = new DateTime('first day of this month');
+        $feedItemOperations []= self::createFeedItemOperation(
```
Comment (fiboknacky) [inline examples/AdvancedOperations/AddAdCustomizer.php]: This is weird.
It should be "array[] =" not "array []="

File: examples/AdvancedOperations/AddAdCustomizer.php [hunk 6]
```diff
@@ -0,0 +1,529 @@
...
+
+        $feedItemResourceNames = [];
+        // Displays the results.
+        foreach ($response->getResults() as $result) {
+            printf(
+                "Created feed item with resource name '%s'.%s",
+                $result->getResourceName(),
+                PHP_EOL
+            );
+            $feedItemResourceNames []= $result->getResourceName();
```
Comment (fiboknacky) [inline examples/AdvancedOperations/AddAdCustomizer.php]: Ditto.

File: examples/AdvancedOperations/AddAdCustomizer.php [hunk 7]
```diff
@@ -0,0 +1,529 @@
...
+    }
+
+  /**
+   * Restricts the feed items to work only with a specific ad group; this prevents the feed items
+   * from being used elsewhere and makes sure they are used only for customizing a specific ad
+   * group.
+   *
+   * @param GoogleAdsClient googleAdsClient the Google Ads API client
+   * @param int $customerId the customer ID
+   * @param array adGroupIds the ad group IDs to bind the feed items to.
```
Comment (fiboknacky) [inline examples/AdvancedOperations/AddAdCustomizer.php]: No "dot" after @param and @return

File: examples/AdvancedOperations/AddAdCustomizer.php [hunk 8]
```diff
@@ -0,0 +1,529 @@
...
+            // Issues a mutate request to add the feed item target.
+            $feedItemTargetServiceClient = $googleAdsClient->getFeedItemTargetServiceClient();
+            $feedItemTargetResponse = $feedItemTargetServiceClient->mutateFeedItemTargets(
+                $customerId,
+                [$feedItemTargetOperation]
+            );
+
+            $feedItemTargetResourceName =
+                $feedItemTargetResponse->getResults()[0]->getResourceName();
+            printf("Added feed item target named '%s'.%s", $feedItemTargetResourceName, PHP_EOL);
```
Comment (fiboknacky) [inline examples/AdvancedOperations/AddAdCustomizer.php]: "Added feed item target with resource name"

File: examples/AdvancedOperations/AddAdCustomizer.php [hunk 9]
```diff
@@ -0,0 +1,529 @@
...
+        }
+    }
+
+    /**
+     * Creates expanded text ads that use the ad customizer feed to populate the placeholders.
+     *
+     * @param GoogleAdsClient googleAdsClient the Google Ads API client.
```
Comment (fiboknacky) [inline examples/AdvancedOperations/AddAdCustomizer.php]: No dot at the end please.
Please check this throughout your file.

File: examples/AdvancedOperations/AddAdCustomizer.php [hunk 10]
```diff
@@ -0,0 +1,529 @@
...
+        $expandedTextAdInfo = new ExpandedTextAdInfo([
+            "headline_part1" => new StringValue(["value" => "Luxury cruise to {=$feedName.Name}"]),
+            "headline_part2" => new StringValue(["value" => "Only {=$feedName.Price}"]),
+            "description" => new StringValue([
+                "value" => "Offer ends in {=countdown($feedName.Date)}!"
+            ])
+        ]);
+
+        $ad = new Ad([
+            "expanded_text_ad" => $expandedTextAdInfo,
```
Comment (fiboknacky) [inline examples/AdvancedOperations/AddAdCustomizer.php]: Use single quote when possible. Check this throughout your file.

File: examples/AdvancedOperations/AddAdCustomizer.php (hunk 9 above)
Comment (Raibaz) [reply examples/AdvancedOperations/AddAdCustomizer.php]: Done.

File: examples/AdvancedOperations/AddAdCustomizer.php (hunk 8 above)
Comment (Raibaz) [reply examples/AdvancedOperations/AddAdCustomizer.php]: Done.

File: examples/AdvancedOperations/AddAdCustomizer.php (hunk 7 above)
Comment (Raibaz) [reply examples/AdvancedOperations/AddAdCustomizer.php]: Done.

File: examples/AdvancedOperations/AddAdCustomizer.php (hunk 6 above)
Comment (Raibaz) [reply examples/AdvancedOperations/AddAdCustomizer.php]: Done.

File: examples/AdvancedOperations/AddAdCustomizer.php (hunk 5 above)
Comment (Raibaz) [reply examples/AdvancedOperations/AddAdCustomizer.php]: Done.

File: examples/AdvancedOperations/AddAdCustomizer.php (hunk 3 above)
Comment (Raibaz) [reply examples/AdvancedOperations/AddAdCustomizer.php]: Done.

File: examples/AdvancedOperations/AddAdCustomizer.php (hunk 2 above)
Comment (Raibaz) [reply examples/AdvancedOperations/AddAdCustomizer.php]: Done.

File: examples/AdvancedOperations/AddAdCustomizer.php [hunk 11]
```diff
@@ -0,0 +1,529 @@
...
+            );
+        }
+    }
+
+    /**
+     * Creates two different feed items to enable two different ad customizations.
+     *
+     * @param GoogleAdsClient googleAdsClient the Google Ads API client
+     * @param int $customerId the customer ID
+     * @param array $feedDetails the names and IDs of feed attributes
```
[the remark this answers is not in the document]
Comment (Raibaz) [reply examples/AdvancedOperations/AddAdCustomizer.php]: Done.

File: examples/AdvancedOperations/AddAdCustomizer.php (hunk 4 above)
Comment (Raibaz) [reply examples/AdvancedOperations/AddAdCustomizer.php]: Done.

File: examples/AdvancedOperations/AddAdCustomizer.php [hunk 12]
```diff
@@ -0,0 +1,529 @@
...
+    ) {
+        $query = "SELECT feed.attributes, feed.name FROM feed WHERE feed.resource_name = " .
+                "'$feedResourceName'";
+
+        $googleAdsServiceClient = $googleAdsClient->getGoogleAdsServiceClient();
+        $response =
+            $googleAdsServiceClient->search($customerId, $query, ['pageSize' => self::PAGE_SIZE]);
+
+        $feed = $response->getIterator()->current()->getFeed();
+        $feedAttributes = $feed->getAttributes();
```
[the remark this answers is not in the document]
Comment (Raibaz) [reply examples/AdvancedOperations/AddAdCustomizer.php]: Done.

File: examples/AdvancedOperations/AddAdCustomizer.php [hunk 13]
```diff
@@ -0,0 +1,529 @@
...
+     * @param string feedResourceName the resource name of the feed
+     * @return array the feed attributes, keyed by attribute name
+     */
+    private static function getFeedAttributes(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        string $feedResourceName
+    ) {
+        $query = "SELECT feed.attributes, feed.name FROM feed WHERE feed.resource_name = " .
+                "'$feedResourceName'";
```
[the remark this answers is not in the document]
Comment (Raibaz) [reply examples/AdvancedOperations/AddAdCustomizer.php]: Done.

File: examples/AdvancedOperations/AddAdCustomizer.php [hunk 14]
```diff
@@ -0,0 +1,529 @@
...
+        // Creates a feed operation for creating a feed.
+        $feedOperation = new FeedOperation();
+        $feedOperation->setCreate($feed);
+
+        // Issues a mutate request to add the feed.
+        $feedServiceClient = $googleAdsClient->getFeedServiceClient();
+        $feedResponse = $feedServiceClient->mutateFeeds($customerId, [$feedOperation]);
+
+        $feedResourceName = $feedResponse->getResults()[0]->getResourceName();
+        printf("Added feed named '%s'.%s", $feedResourceName, PHP_EOL);
```
[the remark this answers is not in the document]
Comment (Raibaz) [reply examples/AdvancedOperations/AddAdCustomizer.php]: Done.

File: examples/AdvancedOperations/AddAdCustomizer.php [hunk 15]
```diff
@@ -0,0 +1,529 @@
...
+        int $customerId,
+        array $adGroupIds
+    ) {
+        if (count($adGroupIds) != self::NUMBER_OF_AD_GROUPS) {
+            throw new \InvalidArgumentException(
+                "Please pass exactly two ad group IDs in the adGroupId parameter."
+            );
+        }
+
+        $feedName = "Ad Customizer example feed " . uniqid();
```
[the remark this answers is not in the document]
Comment (Raibaz) [reply examples/AdvancedOperations/AddAdCustomizer.php]: Done.

File: examples/AdvancedOperations/AddAdCustomizer.php [hunk 16]
```diff
@@ -0,0 +1,529 @@
...
+     * @param int $adGroupId the ad group ID
+     */
+    public static function runExample(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        array $adGroupIds
+    ) {
+        if (count($adGroupIds) != self::NUMBER_OF_AD_GROUPS) {
+            throw new \InvalidArgumentException(
+                "Please pass exactly two ad group IDs in the adGroupId parameter."
```
[the remark this answers is not in the document]
Comment (Raibaz) [reply examples/AdvancedOperations/AddAdCustomizer.php]: Used the constant in the message. Does it look better now? WDYT?

File: examples/AdvancedOperations/AddAdCustomizer.php [hunk 17]
```diff
@@ -0,0 +1,529 @@
...
+{
+    const CUSTOMER_ID = 'INSERT_CUSTOMER_ID_HERE';
+    const AD_GROUP_ID_1 = 'INSERT_AD_GROUP_ID_1_HERE';
+    const AD_GROUP_ID_2 = 'INSERT_AD_GROUP_ID_2_HERE';
+
+    // We're doing only searches by resource_name in this example, we can set page size = 1.
+    const PAGE_SIZE = 1;
+
+    // We're creating two different ad groups to be dynamically populated by the same feed.
+    const NUMBER_OF_AD_GROUPS = 2;
```
[the remark this answers is not in the document]
Comment (Raibaz) [reply examples/AdvancedOperations/AddAdCustomizer.php]: Done.

[the remark this answers is not in the document]
Comment (Raibaz) [reply examples/AdvancedOperations/AddAdCustomizer.php]: (yes using the constant in the const declarations didn't work :))

File: examples/AdvancedOperations/AddAdCustomizer.php [hunk 18]
```diff
@@ -0,0 +1,529 @@
...
+use Google\Ads\GoogleAds\V2\Enums\PlaceholderTypeEnum\PlaceholderType;
+use Google\Ads\GoogleAds\V2\Resources\AdGroupAd;
+use Google\Ads\GoogleAds\V2\Resources\FeedItemTarget;
+use Google\Ads\GoogleAds\V2\Services\AdGroupAdOperation;
+use Google\Ads\GoogleAds\V2\Services\FeedItemTargetOperation;
+use Google\Protobuf\StringValue;
+use Google\ApiCore\ApiException;
+
+/**
+ * This code example adds a page feed to specify precisely which URLs to use with your Dynamic
```
[the remark this answers is not in the document]
Comment (Raibaz) [reply examples/AdvancedOperations/AddAdCustomizer.php]: Done.

File: examples/AdvancedOperations/AddAdCustomizer.php (hunk 16 above)
[the remark this answers is not in the document]
Comment (fiboknacky) [reply examples/AdvancedOperations/AddAdCustomizer.php]: Yes.

File: examples/AdvancedOperations/AddAdCustomizer.php (hunk 13 above)
[the remark this answers is not in the document]
Comment (fiboknacky) [reply examples/AdvancedOperations/AddAdCustomizer.php]: Wrong indentation not fixed.

File: examples/AdvancedOperations/AddAdCustomizer.php [hunk 19]
```diff
@@ -129,11 +129,12 @@ public static function runExample(
     ) {
         if (count($adGroupIds) != self::NUMBER_OF_AD_GROUPS) {
             throw new \InvalidArgumentException(
-                "Please pass exactly two ad group IDs in the adGroupId parameter."
+                "Please pass exactly . " . self::NUMBER_OF_AD_GROUPS .
```
Comment (fiboknacky) [inline examples/AdvancedOperations/AddAdCustomizer.php]: This can be single quote.

File: examples/AdvancedOperations/AddAdCustomizer.php [hunk 20]
```diff
@@ -341,21 +342,20 @@ private function createFeedItems(
...
             $adCustomizerFeedResourceName,
             $adCustomizerFeedAttributes
         );
-
-        $venusDate = DateTime::createFromFormat('d', '15');
-        $feedItemOperations []= self::createFeedItemOperation(
-            "Venus",
-            "$6543.21",
-            date_format($venusDate, "Ymd His"),
+
```
Comment (fiboknacky) [inline examples/AdvancedOperations/AddAdCustomizer.php]: Extra space

File: examples/AdvancedOperations/AddAdCustomizer.php [hunk 21]
```diff
@@ -372,21 +372,21 @@ private function createFeedItems(
...
-     * @param string date the value of the Date attribute.
-     * @param feedResourceName the resource name of the feed.
-     * @param feedAttributes the attributes to be set on the feed.
-     * @return a FeedItemOperation to create a feed item.
+     * @param string name the value of the Name attribute
+     * @param string price the value of the Price attribute
+     * @param string date the value of the Date attribute
+     * @param string adCustomizerFeedResourceName the resource name of the feed
+     * @param array adCustomizerFeedAttributes the attributes to be set on the feed
+     * @return FeedItemOperation to create a feed item
```
Comment (fiboknacky) [inline examples/AdvancedOperations/AddAdCustomizer.php]: FeedItemOperation the feed item operation to create a feed item.

(You need to repeat it)

File: examples/AdvancedOperations/AddAdCustomizer.php [hunk 22]
```diff
@@ -240,22 +241,21 @@ private static function getFeedAttributes(
         int $customerId,
         string $feedResourceName
     ) {
-        $query = "SELECT feed.attributes, feed.name FROM feed WHERE feed.resource_name = " .
-                "'$feedResourceName'";
+        $query = "SELECT feed.attributes, feed.name FROM feed " .
+                 "WHERE feed.resource_name = '$feedResourceName'";
```
Comment (fiboknacky) [inline examples/AdvancedOperations/AddAdCustomizer.php]: Wrong indentation not fixed.

Comment (Raibaz) [reply examples/AdvancedOperations/AddAdCustomizer.php]: Done.

File: examples/AdvancedOperations/AddAdCustomizer.php (hunk 21 above)
Comment (Raibaz) [reply examples/AdvancedOperations/AddAdCustomizer.php]: Done.

File: examples/AdvancedOperations/AddAdCustomizer.php (hunk 20 above)
Comment (Raibaz) [reply examples/AdvancedOperations/AddAdCustomizer.php]: Done.

File: examples/AdvancedOperations/AddAdCustomizer.php (hunk 19 above)
Comment (Raibaz) [reply examples/AdvancedOperations/AddAdCustomizer.php]: Done.

File: examples/AdvancedOperations/AddAdCustomizer.php [hunk 23]
```diff
@@ -0,0 +1,533 @@
...
+        array $adCustomizerFeedAttributes
+    ) {
+        $nameAttributeValue = new FeedItemAttributeValue([
+            'feed_attribute_id' => $adCustomizerFeedAttributes['Name'],
+            'string_value' => new StringValue(['value' => $name])
+        ]);
+
+        $priceAttributeValue = new FeedItemAttributeValue([
+            'feed_attribute_id' => $adCustomizerFeedAttributes['Price'],
+            'string_value' => new StringValue(['value' => $price])
```
[the remark this answers is not in the document]
Comment (Raibaz) [reply examples/AdvancedOperations/AddAdCustomizer.php]: Nope, the price attribute is set to `AdCustomizerPlaceholderField::PRICE` in the field mapping but the value itself is a string_value.

File: examples/AdvancedOperations/AddAdCustomizer.php [hunk 24]
```diff
@@ -0,0 +1,533 @@
...
+            $feedItemResourceNames[] = $result->getResourceName();
+        }
+
+        return $feedItemResourceNames;
+    }
+
+    /**
+     * Creates a FeedItemOperation.
+     *
+     * @param string name the value of the Name attribute
```
[the remark this answers is not in the document]
Comment (Raibaz) [reply examples/AdvancedOperations/AddAdCustomizer.php]: Not sure about this, the attribute names are capitalized so I think they should stay capitalized in the PHPDoc too.

File: examples/AdvancedOperations/AddAdCustomizer.php [hunk 25]
```diff
@@ -0,0 +1,533 @@
...
+        }
+    }
+
+    /**
+     * Creates two different feed items to enable two different ad customizations.
+     *
+     * @param GoogleAdsClient googleAdsClient the Google Ads API client
+     * @param int $customerId the customer ID
+     * @param array $feedDetails the names and IDs of feed attributes
+     * @param string $dsaPageUrlLabel the label for the DSA page URLs
```
[the remark this answers is not in the document]
Comment (Raibaz) [reply examples/AdvancedOperations/AddAdCustomizer.php]: Done.

File: examples/AdvancedOperations/AddAdCustomizer.php [hunk 26]
```diff
@@ -0,0 +1,533 @@
...
+            $googleAdsServiceClient->search($customerId, $query, ['pageSize' => self::PAGE_SIZE]);
+
+        $feed = $response->getIterator()->current()->getFeed();
+        $feedDetails = [];
+        printf(
+            "Found the following attributes for feed with name %s:%s",
+            $feed->getNameUnwrapped(),
+            PHP_EOL
+        );
+        foreach ($$feed->getAttributes() as $feedAttribute) {
```
[the remark this answers is not in the document]
Comment (Raibaz) [reply examples/AdvancedOperations/AddAdCustomizer.php]: Nope, it was a mistake, thanks for spotting!

File: examples/AdvancedOperations/AddAdCustomizer.php [hunk 27]
```diff
@@ -0,0 +1,533 @@
...
+        if (count($adGroupIds) != self::NUMBER_OF_AD_GROUPS) {
+            throw new \InvalidArgumentException(
+                'Please pass exactly ' . self::NUMBER_OF_AD_GROUPS .
+                ' ad group IDs in the adGroupId parameter.'
+            );
+        }
+
+        $feedName = 'Ad Customizer example feed ' . uniqid();
+
+        $adCustomizerFeedResourceName = self::createAdCustomizerFeed(
```
[the remark this answers is not in the document]
Comment (Raibaz) [reply examples/AdvancedOperations/AddAdCustomizer.php]: Done.

File: examples/AdvancedOperations/AddAdCustomizer.php [hunk 28]
```diff
@@ -0,0 +1,533 @@
...
+    // We're doing only searches by resource_name in this example, we can set page size = 1.
+    const PAGE_SIZE = 1;
+
+    public static function main()
+    {
+        // Either pass the required parameters for this example on the command line, or insert them
+        // into the constants above.
+        $options = (new ArgumentParser())->parseCommandArguments([
+            ArgumentNames::CUSTOMER_ID => GetOpt::REQUIRED_ARGUMENT,
+            ArgumentNames::AD_GROUP_ID => GetOpt::MULTIPLE_ARGUMENT
```
[the remark this answers is not in the document]
Comment (Raibaz) [reply examples/AdvancedOperations/AddAdCustomizer.php]: There is no AD_GROUP_IDS constant in ArgumentNames, we use AD_GROUP_ID with MULTIPLE_ARGUMENT in cases like this.

File: examples/AdvancedOperations/AddAdCustomizer.php [hunk 29]
```diff
@@ -0,0 +1,533 @@
...
+            );
+        }
+    }
+
+    /**
+     * Runs the example.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the customer ID
+     * @param int $adGroupId the ad group ID
```
[the remark this answers is not in the document]
Comment (Raibaz) [reply examples/AdvancedOperations/AddAdCustomizer.php]: Done.

File: examples/AdvancedOperations/AddAdCustomizer.php [hunk 30]
```diff
@@ -0,0 +1,533 @@
...
+     */
+    public static function runExample(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        array $adGroupIds
+    ) {
+        if (count($adGroupIds) != self::NUMBER_OF_AD_GROUPS) {
+            throw new \InvalidArgumentException(
+                'Please pass exactly ' . self::NUMBER_OF_AD_GROUPS .
+                ' ad group IDs in the adGroupId parameter.'
```
[the remark this answers is not in the document]
Comment (Raibaz) [reply examples/AdvancedOperations/AddAdCustomizer.php]: Done.

File: examples/AdvancedOperations/AddAdCustomizer.php [hunk 31]
```diff
@@ -0,0 +1,533 @@
...
+            '$1234.56',
+            date_format(new DateTime('first day of this month'), 'Ymd His'),
+            $adCustomizerFeedResourceName,
+            $adCustomizerFeedAttributes
+        );
+        
+        $feedItemOperations[] = self::createFeedItemOperation(
+            'Venus',
+            '$6543.21',
+            // Set the date to the 15th of the current month.
```
[the remark this answers is not in the document]
Comment (Raibaz) [reply examples/AdvancedOperations/AddAdCustomizer.php]: What do you mean?

File: examples/AdvancedOperations/AddAdCustomizer.php (hunk 28 above)
[the remark this answers is not in the document]
Comment (PierrickVoulet) [reply examples/AdvancedOperations/AddAdCustomizer.php]: There are some in plural form like [CAMPAIGN_IDS](https://github.com/googleads/google-ads-php/blob/master/examples/Utils/ArgumentNames.php#L35), might be better keeping it consistent

[the remark this answers is not in the document]
Comment (fiboknacky) [reply examples/AdvancedOperations/AddAdCustomizer.php]: +1 Could you use plural form instead?

File: examples/AdvancedOperations/AddAdCustomizer.php (hunk 31 above)
[the remark this answers is not in the document]
Comment (PierrickVoulet) [reply examples/AdvancedOperations/AddAdCustomizer.php]: Sorry it was not that clear, let me clarify.

It is a nit but there are two calls  of the function createFeedItemOperation and you add a comment "Set the date to the XXth of the current month." to the second one. I am suggesting that you do so to the first one instead as it is the first one a reader will read.

File: examples/AdvancedOperations/AddAdCustomizer.php (hunk 24 above)
[the remark this answers is not in the document]
Comment (PierrickVoulet) [reply examples/AdvancedOperations/AddAdCustomizer.php]: I have no strong position here, both seem reasonable

File: examples/AdvancedOperations/AddAdCustomizer.php (hunk 31 above)
[the remark this answers is not in the document]
Comment (fiboknacky) [reply examples/AdvancedOperations/AddAdCustomizer.php]: I would vote against it.
At most, we should do in both places.
The first one is actually straightforward so we shouldn't need any code comment if it's obvious.

The second one is more tricky and I find myself not sure 100% what that means. 
That's why I asked to put a comment only in the second place.

File: examples/AdvancedOperations/AddAdCustomizer.php (hunk 28 above)
[the remark this answers is not in the document]
Comment (Raibaz) [reply examples/AdvancedOperations/AddAdCustomizer.php]: Done.

**Pull request merged**